Producer/Producer.php

```diff
@@ -120,6 +120,16 @@ public function publishToSelf(Message $message) {
 
     /**
      * @param Message $message
+     * @param $queueName
+     * @throws InvalidExchangeConfigurationException
+     */
+    public function publishToQueue(Message $message, $queueName) {
+        $amqpMessage = $this->encodeAndGetAMQPMessage($message);
+        $this->getChannel()->basic_publish($amqpMessage, "", $queueName);
```

Comment:
Does the channel need to come from the message like `$channel = $message->getQueue()->getChannel();`?